pytorch_lightning/trainer/training_loop.py

```diff
@@ -53,17 +52,17 @@ def __init__(
         self.running_loss = TensorRunningAccum(window_length=20)
         self._curr_step_result = None
         self._cur_grad_norm_dict = None
-        self._multiple_trainloader_mode = multiple_trainloader_mode
         self._skip_backward = False
-        self.trainer._multiple_trainloader_mode = multiple_trainloader_mode
         self._optimizer_freq_cumsum = None
+        self._hiddens = None
 
         self.global_step = 0
         self.current_epoch = 0
         self.trainer.should_stop = False
 
         self.total_batch_idx = 0
         self.batch_idx = 0
+        self.split_idx = None
```

Comment:
- is this meant to be writable from the outside? should these be made available as properties instead? 
- while going through this, could you add a comment for what split_idx means for contributors who go through this code later?

Reply:
no but for now I follow the above pattern and the major goal is to define this state strictly on the loop, not the trainer anymore.

Reply:
> while going through this, could you add a comment for what split_idx means for contributors who go through this code later?

yes

Reply:
okay added comments